.github/workflows/advanced.yml

```diff
@@ -34,111 +36,150 @@ jobs:
 
     env:
       BINUTILS_REPO: Windows-on-ARM-Experiments/binutils-woarm64
-      BINUTILS_BRANCH: ${{ github.event.inputs.binutils_branch }}
+      BINUTILS_BRANCH: ${{ github.event.inputs.binutils_branch || 'woarm64' }}
       BINUTILS_VERSION: binutils-master
       GCC_REPO: Windows-on-ARM-Experiments/gcc-woarm64
-      GCC_BRANCH: ${{ github.event.inputs.gcc_branch }}
+      GCC_BRANCH: ${{ github.event.inputs.gcc_branch || 'woarm64' }}
       GCC_VERSION: gcc-master
       MINGW_REPO: Windows-on-ARM-Experiments/mingw-woarm64
-      MINGW_BRANCH: ${{ github.event.inputs.mingw_branch }}
+      MINGW_BRANCH: ${{ github.event.inputs.mingw_branch || 'woarm64' }}
       MINGW_VERSION: mingw-w64-master
       TARGET: ${{ matrix.target }}
       CRT: ${{ matrix.crt }}
+      TOOLCHAIN_NAME: ${{ matrix.target }}-${{ matrix.crt }}
       TOOLCHAIN_PACKAGE_NAME: ${{ matrix.target }}-${{ matrix.crt }}-toolchain.tar.gz
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
           path: ${{ github.workspace }}
-        
+
+      - name: Get binutils SHA
+        id: binutils-sha
+        run: |
+          .github/scripts/get-sha.sh ${{ env.BINUTILS_REPO }} ${{ env.BINUTILS_BRANCH }}
+
+      - name: Get GCC SHA
+        id: gcc-sha
+        run: |
+          .github/scripts/get-sha.sh ${{ env.GCC_REPO }} ${{ env.GCC_BRANCH }}
+
+      - name: Get MinGW SHA
+        id: mingw-sha
+        run: |
+          .github/scripts/get-sha.sh ${{ env.MINGW_REPO }} ${{ env.MINGW_BRANCH }}
+
+      - name: Cache toolchain
```

Comment:
It would be great to have binutils cache separate to speed up gcc build with changes. which would be the most common case in the nearest contributions. For now it is fine, just an idea for next iteration on cache.

Reply:
I agree, let's do it as a separate PR.

Reply:
I agree, let's do it as a separate PR.